cmake/ConfigureCompiler.cmake

```diff
@@ -41,6 +41,7 @@ endif()
 add_compile_options(-DCMAKE_BUILD)
 add_compile_definitions(BOOST_ERROR_CODE_HEADER_ONLY BOOST_SYSTEM_NO_DEPRECATED)
 
+set(THREADS_PREFER_PTHREAD_FLAG ON)
```

Comment:
This is a (temporary?) fix I added to allow the linker to successfully link pthreads for the unit test binaries with `ld`. It causes `ld` to use `-pthread` when linking instead of `-lpthread` ([docs](https://cmake.org/cmake/help/latest/module/FindThreads.html#variable:THREADS_PREFER_PTHREAD_FLAG)). I was running into an issue where the thread library was being linked before another library and causing a linker failure, and this was the best way I have come up with so far to fix it. Interested in getting help on this from anyone with more expertise with build systems.